db-migration.Dockerfile

```diff
@@ -1,4 +1,15 @@
-FROM liquibase/liquibase:3.10.x
+# Use Liquibase as base image
+FROM liquibase/liquibase:4.25
 
-COPY --chown=liquibase:liquibase changelog ./changelog
-COPY --chown=liquibase:liquibase scripts /scripts
\ No newline at end of file
+# Change to root user for installation
+USER root
+
+# Install Azure CLI
+RUN curl -skL https://aka.ms/InstallAzureCLIDeb | bash
+
+# Copy scripts and migration changelog files
+COPY --chown=liquibase:liquibase --chmod=755 changelog ./changelog
+COPY --chown=liquibase:liquibase --chmod=755 scripts ./scripts
+
+# Get AAD access token and run database migration
+CMD ["sh", "-c", "az login --service-principal --username $AZURE_CLIENT_ID --federated-token $(cat $AZURE_FEDERATED_TOKEN_FILE) --tenant $AZURE_TENANT_ID; ACCESS_TOKEN=$(az account get-access-token --resource-type oss-rdbms --query 'accessToken' --output tsv); export ACCESS_TOKEN; cd scripts/migration; sh database-up $ACCESS_TOKEN; exit 0;"]
```

Comment:
Do we have to do this for all services? Can we have it as a common logic?

Reply:
@ppotturi  We could have this taken out as a separate script file. But we might still need to add that script file to all applications that need migration.

Reply:
As we discussed, We can keep this logic here. But we will investigate if there are any better ways to do it.